### PR TITLE
Add keycloak provider

### PR DESCRIPTION
**Pull request by Ofinka**
Adding integration of keycloak provider based on forgotten PR https://github.com/bitly/oauth2_proxy/pull/366/files 

Solves https://github.com/pusher/oauth2_proxy/issues/127

## How Has This Been Tested?

Tested manually. I set up keycloak and reference it in configuration of oauth2_proxy using this provider.

## 
- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by mannp** [comment]
Excellent, thank you for taking the time to do this :-)

**Comment by syscll** [comment]
LGTM :+1: - @JoelSpeed do we have any additional processes that need completing when adding a new provider?

File: options.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type Options struct {
 	TLSKeyFile      string `flag:"tls-key-file" cfg:"tls_key_file" env:"OAUTH2_PROXY_TLS_KEY_FILE"`
 
 	AuthenticatedEmailsFile  string   `flag:"authenticated-emails-file" cfg:"authenticated_emails_file" env:"OAUTH2_PROXY_AUTHENTICATED_EMAILS_FILE"`
+	KeycloakGroup            string   `flag:"keycloak-group" cfg:"keycloak_group"`
```
Comment (JoelSpeed) [inline options.go]: Please add the relevant `env` tag too for consistency please

**Comment by steakunderscore** [comment]
@Ofinka Do you mind adding yourself to the owners file for future changes? https://github.com/pusher/oauth2_proxy/blob/master/.github/CODEOWNERS#L12

**Comment by mannp** [comment]
Has this one stalled now, as it seemed to be close to getting merged 😃

**Comment by Ofinka** [comment]
Hi guys, I'm sorry but I'm afraid I do not have enough time to be part of another project :( 

Would definitely appreciate if anyone else could take the responsibility.

**Comment by steakunderscore** [comment]
Understood. I think we should merge this as is.

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,6 +51,8 @@ reconfigure their proxies. Please read the Breaking Changes below thoroughly.
 
 ## Changes since v3.2.0
 
+- [#178](https://github.com/pusher/outh2_proxy/pull/178) Add Silence Ping Logging and Exclude Logging Paths flags (@kskewes)
```
Comment (JoelSpeed) [inline CHANGELOG.md]: This looks like a merge error, these two entries should not be here

**Comment by mleklund** [comment]
Would love to see this move along.

**Comment by syscll** [comment]
If we can fix @JoelSpeed's comment above, then I can get this merged.

**Comment by mleklund** [comment]
The author of the pull request has not responded since August 19, not sure when or if they will.

File: CHANGELOG.md (hunk 2 above)
Comment (syscll) [reply CHANGELOG.md]: I have fixed this in order to get it merged.

**Comment by syscll** [comment]
I have fixed the CHANGELOG error. Just need to get it re-reviewed by @JoelSpeed or @steakunderscore

**Comment by syscll** [comment]
Thank you and congrats on your first contribution! :tada:

**Comment by mleklund** [comment]
Thanks for your help!!

**Pull request merged**